### ISSUE TITLE
Type Annotation - if param has default value, don't complain

### ISSUE THREAD

**Issue by markwpearce** [rokucommunity/bslint#131]
Example:

```
sub testTypeAnnotation(doOption = false) 
end sub
```

Currently, this would fail the `type-annotation` rule, but `doOption` param would be known as a `boolean`, and adding a type to it is superfluous

Perhaps we could add a `args-allow-default` and `all-allow-default` options for the `type-annotation` rule
